### PR TITLE
Expression UI handling

### PR DESCRIPTION
**Pull request by klausi**
See https://www.drupal.org/node/2530694

### CODE REVIEW

File: modules/rules_ui/rules_ui.routing.yml [hunk 1]
```diff
@@ -79,4 +79,12 @@ entity.rules_component.delete_form:
...
\ No newline at end of file
+    _permission: 'administer rules+administer rules components'
+
+rules_ui.condition.add:
+  path: '/admin/config/workflow/rules/reactions/{rules_reaction_rule}/add/{rules_condition_offset}'
+  requirements:
+    _permission: 'administer rules+administer rules reactions'
+  defaults:
+    _title: 'Add condition'
+    _form: '\Drupal\rules\From\AddConditionForm'
```
Comment (fago) [inline modules/rules_ui/rules_ui.routing.yml]: From? :)

File: src/Annotation/RulesExpression.php [hunk 2]
```diff
@@ -39,4 +39,11 @@ class RulesExpression extends Plugin {
    */
   public $type = '';
 
+  /**
+   * The clss name of the form for displaying/editing this expression.
+   *
+   * @var string
+   */
+  public $form;
```
Comment (fago) [inline src/Annotation/RulesExpression.php]: class name -> form_class maybe? what does core here, ommit class name?

**Comment by mark-casias** [comment]
After applying patch, I am receiving 

`<em class="placeholder">Drupal\Component\Plugin\Exception\PluginNotFoundException</em>: The &quot;&quot; plugin does not exist. in <em class="placeholder">Drupal\Core\Plugin\DefaultPluginManager-&gt;doGetDefinition()</em> (line <em class="placeholder">57</em> of <em class="placeholder">/Library/WebServer/Documents/ksu/core/lib/Drupal/Component/Plugin/Discovery/DiscoveryTrait.php</em>).` 

When I try to edit a reaction rule.

**Comment by stephenpurkiss** [comment]
Hi - here's a couple of fixes. I couldn't figure out for the life of me how to push using git, I did try :dancer: 

diff --git a/modules/rules_ui/rules_ui.routing.yml b/modules/rules_ui/rules_ui.routing.yml
index 332e66e..3e4f811 100755
--- a/modules/rules_ui/rules_ui.routing.yml
+++ b/modules/rules_ui/rules_ui.routing.yml
@@ -87,4 +87,4 @@ rules_ui.condition.add:
     _permission: 'administer rules+administer rules reactions'
   defaults:
     _title: 'Add condition'
-    _form: '\Drupal\rules\From\AddConditionForm'
-    _form: '\Drupal\rules\Form\AddConditionForm'
  diff --git a/src/Form/AddConditionForm.php b/src/Form/AddConditionForm.php
  index b41200c..7efacbc 100644
  --- a/src/Form/AddConditionForm.php
  +++ b/src/Form/AddConditionForm.php
  @@ -9,6 +9,7 @@ namespace Drupal\rules\Form;
  
  use Drupal\Core\Form\FormBase;
  use Drupal\Core\Form\FormStateInterface;
  +use Symfony\Component\DependencyInjection\ContainerInterface;
  
  class AddConditionForm extends FormBase {

**Comment by klausi** [comment]
merged.

**Pull request closed without merging**